src/main.rs

```diff
@@ -15,30 +15,83 @@ use std::path::PathBuf;
 pub type Cost = NotNan<f64>;
 pub const INFINITY: Cost = unsafe { NotNan::new_unchecked(std::f64::INFINITY) };
 
-fn main() {
-    env_logger::init();
+#[derive(PartialEq, Eq)]
```

Comment:
Can you move the testing code to a new module? Maybe `src/test.rs` or something?

Reply:
Yes, no worries, I've committed that change.